### PR TITLE
docs: add MNIST magnitude pruning tutorial 

### PR DESCRIPTION
**Pull request by u-simha**
This PR:
- Adds a MNIST magnitude pruning tutorial notebook
- Updates the sidebar to surface the Pruning documentation

### Testing

- [x] Execute notebook locally

### CODE REVIEW

File: docs/src/examples/toy_models.md [hunk 1]
```diff
@@ -8,4 +8,5 @@ These tutorials demonstrate model compression techniques using simple toy models
 /tutorials/mnist_quantization
 /tutorials/mnist_palettization
 /tutorials/mnist_palettization_and_activation_quantization
+/tutorials/mnist_pruning
```
Comment (guru-desh) [inline docs/src/examples/toy_models.md:11]: Can you confirm that `make test-tutorials` is executing this notebook?

Comment (guru-desh) [reply docs/src/examples/toy_models.md:11]: Can you also run `make docs` and check that this appears when we build docs?

File: docs/src/tutorials/mnist_pruning.ipynb [hunk 2]
```diff
@@ -0,0 +1,737 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cell-23",
+   "metadata": {},
+   "source": [
+    "Magnitude pruning sparsifies a model by zeroing out the smallest-magnitude weights, up to a `target_sparsity`. Post-training pruning applies the full sparsity in a single shot during the `prepare()` call — no calibration data or fine-tuning is required.\n",
+    "\n",
+    "Unless a model's weights are already close to zero, post-training pruning will usually degrade accuracy. It's most useful as a quick way to see the effect of sparsity before committing to a fine-tuning workflow."
```
Comment (guru-desh) [inline docs/src/tutorials/mnist_pruning.ipynb:424]: "It's most useful as a quick way to see the effect of sparsity before committing to a fine-tuning workflow."

What does "effect of sparsity" mean? From what I thought, it's so that we can test performance of a sparse model on-device to see performance gains. "effect of sparsity" seems vague because it could also refer to accuracy which doesn't make sense since it would degrade accuracy but fine-tuning would bring up the accuracy back again

Comment (u-simha) [reply docs/src/tutorials/mnist_pruning.ipynb:424]: It means the latter - and is consistent with what we mention for post-training quantization and post-training palettization: 
> incorporate the effect of quantization
https://apple.github.io/coreai-optimization/quantization/advanced.html#qat-api-overview-with-custom-schedule

Since we are in the PyTorch world, it would mean the impact on the accuracy. Unless we do structured sparsity, and we have the ops that allow for latency improvements with sparsity, the effects of sparsity for on-device performance is not a simple equation (depends on whether the model is memory bound or not). 

Therefore any reference to effects of <sparsity | quantization | palettization> across the docs have consistently meant accuracy degradation.

**Review by guru-desh — APPROVED** [review]

File: docs/src/examples/toy_models.md (hunk 1 above)
Comment (u-simha) [reply docs/src/examples/toy_models.md:11]: `make test-tutorials`:
```================================ 5 passed, 2 warnings in 153.34s (0:02:33) ================================
```
`make docs`: 
Screenshot 2026-07-15 at 5 36 29 PM
